openedx/features/content_tagging/rest_api/v1/filters.py

```diff
@@ -0,0 +1,45 @@
+"""
+API Filters for content tagging org
+"""
+
+from django.db.models import Exists, OuterRef, Q
+from organizations.models import Organization
+from rest_framework.filters import BaseFilterBackend
+
+from ...models import TaxonomyOrg
+from ...rules import is_taxonomy_admin, is_taxonomy_user_for_org
+
+
+class UserOrgFilterBackend(BaseFilterBackend):
+    """
+    Filter taxonomies based on the user's organization.
+    """
+
+    def filter_queryset(self, request, queryset, _):
+        if is_taxonomy_admin(request.user):
+            return queryset
+
+        return queryset.filter(enabled=True).filter(
+            Q(
+                Exists(
+                    TaxonomyOrg.objects.filter(
+                        taxonomy=OuterRef("pk"),
+                        rel_type=TaxonomyOrg.RelType.OWNER,
+                        org=None,
+                    )
+                )
+            )
+            | Q(
+                Exists(
+                    TaxonomyOrg.objects.filter(
+                        taxonomy=OuterRef("pk"),
+                        rel_type=TaxonomyOrg.RelType.OWNER,
+                        org__short_name__in=[
+                            org.short_name
+                            for org in Organization.objects.all()
+                            if is_taxonomy_user_for_org(request.user, org)
+                        ],
+                    )
+                )
+            )
+        )
```

Comment:
We only need the `enabled` filter here -- the rest of this org checking can go.

```suggestion
        return queryset.filter(enabled=True)
```